crates/js_api/src/gui/select_tokens.rs

```diff
@@ -134,6 +135,57 @@ impl DotrainOrderGui {
         }
         Ok(true)
     }
+
+    #[wasm_export(js_name = "getAllTokens", unchecked_return_type = "TokenInfo[]")]
+    pub async fn get_all_tokens(&self) -> Result<Vec<TokenInfo>, GuiError> {
+        let network_key = self.get_network_key()?;
+        let tokens = self.dotrain_order.orderbook_yaml().get_tokens()?;
+        let network = self
+            .dotrain_order
+            .orderbook_yaml()
+            .get_network(&network_key)?;
+
+        let mut fetch_futures = Vec::new();
+        let mut results = Vec::new();
+
+        for (_, token) in tokens
+            .into_iter()
+            .filter(|(_, token)| token.network.key == network_key)
+        {
+            if token.decimals.is_none() || token.label.is_none() || token.symbol.is_none() {
+                let erc20 = ERC20::new(network.rpc.clone(), token.address);
+                fetch_futures.push(async move {
+                    let token_info = erc20.token_info(None).await?;
```

Comment:
hm what if there's 500 tokens in the list? the rpc will for sure ratelimit - do we need token decimals for the purposes of what we're doing?

Reply:
Hmm, could we at least leave an option to not fetch them eagerly - if they're needed for approvals can't we just fetch them then if still missing?

Reply:
besides decimals, one other part we need the token information is to show the token name and symbol in the dropdown ui. we can fetch decimals during calldata generation but without the token info that shows up on the ui, the dropdown feature does not make sense

Reply:
to not get rate limited, we limit the amount of requests we send by the way. it's 5 requests max in flight

Reply:
yeah like i said if there are missing fields in the response in each token data when we are fetching remote tokens from given url, an error is thrown saying that the response could not be decoded. this means it means the tokens we get are filled with the data we need

this missing data is only possible during manual typing